pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -88,7 +87,8 @@ public ImmutableSegmentImpl(SegmentDirectory segmentDirectory, SegmentMetadataIm
     _segmentMetadata = segmentMetadata;
     _indexContainerMap = columnIndexContainerMap;
     _starTreeIndexContainer = starTreeIndexContainer;
```

Comment:
Should be fine because map is keyed by String (which caches hashcode) and we know map size ahead of time.